src/window/advisors.h

```diff
@@ -4,14 +4,19 @@
 #include "city/constants.h"
 #include "input/mouse.h"
 
+typedef struct {
+    int text_id;
+    int translation_key;
+} advisor_tooltip_result;
+
 typedef struct {
     /**
      * @return height of the advisor in blocks of 16px
      */
     int (*draw_background)(void);
     void (*draw_foreground)(void);
     int (*handle_mouse)(const mouse *m);
-    int (*get_tooltip_text)(void);
+    advisor_tooltip_result(*get_tooltip_text)(void);
```

Comment:
Instead of making the callback a function returning an `advisor_tooltip_result` struct, please send a pointer to that struct as a parameter:

```suggestion
    void (*get_tooltip_text)(advisor_tooltip_result *);
```